_posts/2021-08-15-using-http-bridge-as-a-kubernetes-sidecar.md

```diff
@@ -0,0 +1,408 @@
+
+```
+curl -X POST http://localhost:8080/consumers/my-group \
+     -H 'Content-Type: application/vnd.kafka.v2+json' \
+     -d '{
+           "name": "my-consumer",
+           "auto.offset.reset": "earliest",
+           "format": "json",
+           "enable.auto.commit": true,
+           "fetch.min.bytes": 512,
+           "consumer.request.timeout.ms": 30000
+         }'
+```
+
+Then we subscribe to the topic `my-topic`:
+
+```
+curl -X POST http://localhost:8080/consumers/my-group/instances/my-consumer/subscription \
+     -H 'Content-Type: application/vnd.kafka.v2+json' \
+     -d '{
+           "topics": [
+             "my-topic"
+           ]
+         }'
+```
+
+And then we can just consume the messages:
+
+```
+curl -X GET http://localhost:8080/consumers/my-group/instances/my-consumer/records \
+     -H 'Accept: application/vnd.kafka.json.v2+json'
+```
+
+You might need to call this command multiple times while the consumer initializes.
+But eventually you should get the message we have sent before:
+
+```json
+[{"topic":"my-topic","key":null,"value":"Hello World!","partition":0,"offset":0}]
+```
+
+And that is it.
+We have now working pod with the HTTP bridge as a sidecar.
+For more details about the `curl` commands and how to use the bridge, you can check the [HTTP Bridge documentation](https://strimzi.io/docs/operators/latest/full/using.html#assembly-kafka-bridge-quickstart-str) and its [API reference](https://strimzi.io/docs/bridge/latest/).
+
+## More complex example
+
+The previous example with manually executed `curl` commands shows how to deploy and use the bridge sidecar.
+But it is of course rather simple.
+If you are interested in something more complex, we have another example for you.
+It is using our [example HTTP clients](https://github.com/strimzi/client-examples) coupled with the bridge sidecar.
+It creates two deployments - one for producing and one for consuming messages.
+The full YAML of this example can be found [here](https://gist.github.com/scholzj/8902c6bb3de9fa7fbc6e75ee22ee870e).
+
+## When to use it
+
+This blog post shows an alternative way how to use the Strimzi HTTP Bridge.
+When running as a sidecar, a separate instance of the bridge runs with every instance of your application.
+That way, it might consume more resources compared to using a single central instance of the bridge deployed by the operator.
+
+But running it as a sidecar has also some advantages.
+Since the bridge is used only locally, you have less worries about securing it.
+You do not need any API Gateway or Proxy to secure it.
+You have also less worries about scaling the bridge.
+Each instance of your application will have its own bridge instance and will behave exactly as if it used a built in Kafka client.
+Even when using it to consume messages, it will have exactly the same properties as native Kafka client including message ordering or partitions assignments.
```

Comment:
```suggestion
Even when using it to consume messages, it will have exactly the same properties as a native Kafka client including message ordering or partitions assignments.
```